chia/farmer/farmer.py

```diff
@@ -547,25 +546,30 @@ def get_public_keys(self):
     def get_private_keys(self):
         return self._private_keys
 
-    async def get_reward_targets(self, search_for_private_key: bool) -> Dict:
+    async def get_reward_targets(self, search_for_private_key: bool, max_ph_to_search: int = 500) -> Dict:
         if search_for_private_key:
             all_sks = await self.get_all_private_keys()
-            stop_searching_for_farmer, stop_searching_for_pool = False, False
-            for i in range(500):
-                if stop_searching_for_farmer and stop_searching_for_pool and i > 0:
+            have_farmer_sk, have_pool_sk = False, False
+            search_addresses: List[bytes32] = [self.farmer_target, self.pool_target]
+            for sk, _ in all_sks:
+                found_dict: Dict[bytes32, bool] = match_address_to_sk(sk, search_addresses, max_ph_to_search)
```

Comment:
minor: A set would be a bit tidier to work with, but the dictionary approach works as well.

Reply:
yea, good idea